Circuit/Components/VacuumTubes/Pentode.cs

```diff
@@ -0,0 +1,113 @@
+﻿using ComputerAlgebra;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+
+namespace Circuit.Components
+{
+    [Category("Vacuum Tubes")]
+    [DisplayName("Pentode")]
+    public class Pentode : Component
+    {
+        private Terminal _plate, _grid, _grid2, _cathode;
+
+        private double _mu = 10.7;
+        [Serialize, Category("Koren"), Description("Voltage gain.")]
+        public double Mu { get { return _mu; } set { _mu = value; NotifyChanged(nameof(Mu)); } }
+
+        private double _kg1 = 1672.0;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kg1 { get { return _kg1; } set { _kg1 = value; NotifyChanged(nameof(Kg1)); } }
+
+        private double _kg2 = 4500;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kg2 { get { return _kg2; } set { _kg2 = value; NotifyChanged(nameof(Kg2)); } }
+
+        private double _kp = 41.16;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kp { get { return _kp; } set { _kp = value; NotifyChanged(nameof(Kp)); } }
+
+        private double _kvb = 12.7;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Kvb { get { return _kvb; } set { _kvb = value; NotifyChanged(nameof(Kvb)); } }
+
+        private double _ex = 1.310;
+        [Serialize, Category("Koren"), Browsable(true)]
+        public double Ex { get { return _ex; } set { _ex = value; NotifyChanged(nameof(Ex)); } }
+
+        public Pentode()
+        {
+            _plate = new Terminal(this, "P");
+            _grid = new Terminal(this, "G");
+            _grid2 = new Terminal(this, "G2");
+            _cathode = new Terminal(this, "K");
+        }
+
+        public override IEnumerable<Terminal> Terminals
+        {
+            get
+            {
+                yield return _plate;
+                yield return _grid;
+                yield return _grid2;
+                yield return _cathode;
+            }
+        }
+
+        protected internal override void LayoutSymbol(SymbolLayout Sym)
+        {
+            Sym.AddTerminal(_plate, new Coord(0, 20), new Coord(0, 8));
+            Sym.AddWire(new Coord(-10, 8), new Coord(10, 8));
+
+            Sym.AddTerminal(_grid, new Coord(-20, 0), new Coord(-12, 0));
+            Sym.AddTerminal(_grid2, new Coord(20, 4), new Coord(12, 4));
```

Comment:
How do you wire this terminal?
I had to modify the layout with `Sym.AddTerminal(_grid2, new Coord(20, 10), new Coord(16, 4), new Coord(12, 4));` so the terminal match the grid alignment.

Reply:
By dragging wire through ;) I've recently changed this symbol, but I was unable to make it look nice without changing the grid snap to 5 points. Now it looks and works fine, but I'm unsure if I can leave it like that (@dsharlet?). I didn't touched grid lines, because it felt little bit too cluttered with 5 point spacing. Maybe it would be a good idea to make it user-selectable like in most schematic design software?

Reply:
I've long regretted not making this grid spacing a power of 2 :) Unfortunately at this point there aren't many options for changing it, it probably needs to be a factor of the existing grid spacing (10) to avoid breaking all the existing schematics that exist. I think changing it to 5 is probably OK. Making it a user option is reasonable too, the problem is I'd really change a bunch of stuff about the components if the spacing were a power of 2, so I'm not sure how well it will work to make this user selectable, it might just not make sense to use anything other than 5 or 10.

I'll try it out with 5.